instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py

```diff
@@ -0,0 +1,174 @@
+import functools
+import typing as t
+
+from aiohttp.web import BaseRequest, RequestHandler, StreamResponse
+
+from opentelemetry import context, propagate, trace
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import (
+    extract_attributes_from_object,
+    http_status_to_status_code,
+)
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.util.http import (
+    ExcludeList,
+    get_excluded_urls,
+    get_traced_request_attrs,
+)
+
+from .version import __version__
+
+RequestHook = t.Callable[[trace.Span, BaseRequest], t.Any]
+ResponseHook = t.Callable[[trace.Span, BaseRequest, StreamResponse], t.Any]
+
+
+_CONFIG_PREFIX = "AIOHTTP_SERVER"
+_CONFIG_TRACED_REQUEST_ATTRS = get_traced_request_attrs(_CONFIG_PREFIX)
+_CONFIG_EXCLUDED_URLS = get_excluded_urls(_CONFIG_PREFIX)
+_SPAN_KEY = "opentelemetry-instrumentor-aiohttp-server.span_key"
+
+
+def _span_name(request: BaseRequest) -> str:
+    return f"HTTP {request.method.upper()}"
```

Comment:
Endpoint without variables would be nice ex: `/users/{user_id}` but if we can't get that this is fine.

Reply:
Due to the architecture of the `aiohttp server` at this point `route` is unfortunately not available (because only `BaseRequest` is passed). It would have been available if we used a middleware (`Request` is passed), however that's a trade off between different aspects (see the initial comment).